### PR TITLE
document: advanced search config and endpoint

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Adds configuration for enabled/disabled advanced search.
* Adds a new field in the document schema to group
  call number for holdings and items.
* Adds a new endpoint for advanced search config.

⚠️  update ES mapping and reindex ES documents

### CODE REVIEW

**Comment by PascalRepond** [comment]
Could you add migration instructions? Update mappings, reindex documents?

**Review by zannkukai — APPROVED** [review]

File: rero_ils/modules/documents/mappings/v7/documents/document-v0.0.1.json [hunk 1]
```diff
@@ -2135,6 +2139,9 @@
           }
         }
       },
+      "call_numbers": {
+        "type": "text"
```
Comment (jma) [inline rero_ils/modules/documents/mappings/v7/documents/document-v0.0.1.json:2143]: Do we need a subproperty of type raw?

**Review by jma — APPROVED** [review]

**Comment by PascalRepond** [comment]
## Tests

- [ ] REP: The new search bar position is weird. Is it possible to move it above the results number and filter buttons:
image
- [ ] REP: Field selector dropdown values should be fixed ordered (order in the [US](https://tree.taiga.io/project/rero21-reroils/us/2652))

**Review by lauren-d — APPROVED** [review]

File: rero_ils/config.py [hunk 2]
```diff
@@ -3632,3 +3635,108 @@ def _(x):
...
+    },
+    {
+        'label': 'Country',
+        'value': 'country',
+        'field': 'provisionActivity.country',
+    },
+    {
+        'label': 'Canton',
+        'value': 'canton',
+        'field': 'provisionActivity.canton',
```
Comment (PascalRepond) [inline rero_ils/config.py:3665]: `provisionActivity.place.canton`

File: rero_ils/config.py [hunk 3]
```diff
@@ -3632,3 +3635,108 @@ def _(x):
...
+    },
+    {
+        'label': 'Provision activity statement',
+        'value': 'provisionActivityStatement',
+        'field': 'provisionActivity.statement.label',
```
Comment (PascalRepond) [inline rero_ils/config.py:3670]: `provisionActivity._text.value`

File: rero_ils/config.py [hunk 4]
```diff
@@ -3632,3 +3635,108 @@ def _(x):
...
+    },
+    {
+        'label': 'Document local fields',
+        'value': 'documentLocalFields',
+        'field': 'local_fields.*',
+    },
+    {
+        'label': 'RDA content type',
+        'value': 'rdaContentType',
+        'field': 'contentmediaCarrier.contentType',
```
Comment (PascalRepond) [inline rero_ils/config.py:3730]: `contentMediaCarrier.contentType`

File: rero_ils/config.py [hunk 5]
```diff
@@ -3632,3 +3635,108 @@ def _(x):
...
+    },
+    {
+        'label': 'RDA media type',
+        'value': 'rdaMediaType',
+        'field': 'contentmediaCarrier.mediaType',
```
Comment (PascalRepond) [inline rero_ils/config.py:3735]: `contentMediaCarrier.mediaType`

File: rero_ils/config.py [hunk 6]
```diff
@@ -3632,3 +3635,108 @@ def _(x):
...
+    },
+    {
+        'label': 'RDA carrier type',
+        'value': 'rdaCarrierType',
+        'field': 'contentmediaCarrier.carrierType',
```
Comment (PascalRepond) [inline rero_ils/config.py:3740]: `contentMediaCarrier.carrierType`

File: rero_ils/config.py [hunk 7]
```diff
@@ -3632,3 +3635,108 @@ def _(x):
...
+    {
+        'label': 'Holdings local fields',
+        'value': 'holdingsLocalFields',
+        'field': 'holdings.local_fields.*',
+    },
+    {
+        'label': 'Classification',
+        'value': 'classification',
+        'field': 'classification.*',
+    },
```
Comment (PascalRepond) [inline rero_ils/config.py:3716]: Order: please put this under all the local fields

File: rero_ils/config.py [hunk 8]
```diff
@@ -3632,3 +3635,108 @@ def _(x):
...
+        'value': 'subjects',
+        'field': 'subjects.entity.*',
+    },
+    {
+        'label': 'Call number',
+        'value': 'callNumber',
+        'field': 'call_numbers',
+    },
+    {
+        'label': 'Holdings local fields',
```
Comment (PascalRepond) [inline rero_ils/config.py:3708]: `'label': 'Local fields (holdings)'`

File: rero_ils/config.py [hunk 9]
```diff
@@ -3632,3 +3635,108 @@ def _(x):
...
+        'value': 'holdingsLocalFields',
+        'field': 'holdings.local_fields.*',
+    },
+    {
+        'label': 'Classification',
+        'value': 'classification',
+        'field': 'classification.*',
+    },
+    {
+        'label': 'Item local fields',
```
Comment (PascalRepond) [inline rero_ils/config.py:3718]: `'label': 'Local fields (items)'`

File: rero_ils/config.py [hunk 10]
```diff
@@ -3632,3 +3635,108 @@ def _(x):
...
+        'value': 'itemLocalFields',
+        'field': 'holdings.items.local_fields.*',
+    },
+    {
+        'label': 'Document local fields',
```
Comment (PascalRepond) [inline rero_ils/config.py:3723]: `'label': 'Local fields (document)'`

**Comment by PascalRepond** [comment]
## Tests (round 2)

- [ ] MD: 500 error if I use double quotes without deselecting the option 'phrase'. Once I get the error I cannot use the advanced search during 5min (local storage). The page is blocked even after a refresh.
- [ ] REP: "build advanced query button" should be always visible
- [ ] REP: When I reload the page after an adavanced search and re-open the advanced query builder, I get an error "ERROR TypeError: this.fieldMappingMap is undefined"

**Review by rerowep — APPROVED** [review]

**Review by zannkukai — APPROVED** [review]

File: rero_ils/config.py [hunk 11]
```diff
@@ -3639,3 +3642,183 @@ def _(x):
...
+    :param: field: the field key.
+    :return: a list of options.
+    """
+    output = [];
+    if field not in [
+        'country', 'canton', 'rdaContentType', 'rdaMediaType',
+        'rdaCarrierType'
+    ]:
+        output.append({'label': _('contains'), 'value': 'contains'})
+    output.append({'label': _('exact'), 'value': 'phrase'})
```
Comment (PascalRepond) [inline rero_ils/config.py:3660]: Fields "isbn", "issn", and "identifier" should only have option "contains". A phrase search makes no sense for these fields.

Comment (Garfield-fr) [reply rero_ils/config.py]: Fixed

**Review by PascalRepond — APPROVED** [review]

**Pull request merged**